packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -81,28 +81,39 @@ async function generateCacheEntry(
   fn: any
 ): Promise<ReadableStream> {
   const temporaryReferences = createServerTemporaryReferenceSet()
-  const [, args] = await decodeReply<any[]>(encodedArguments, serverManifest, {
-    temporaryReferences,
-  })
+
+  const [, args] = await decodeReply<any[]>(
+    encodedArguments,
+    getServerModuleMap(),
+    {
+      temporaryReferences,
+    }
+  )
 
   // Invoke the inner function to load a new result.
   const result = fn.apply(null, args)
 
   let didError = false
   let firstError: any = null
 
-  const stream = renderToReadableStream(result, clientManifest, {
-    environmentName: 'Cache',
-    temporaryReferences,
-    onError(error: any) {
-      // Report the error.
-      console.error(error)
-      if (!didError) {
-        didError = true
-        firstError = error
-      }
-    },
-  })
+  const clientReferenceManifestSingleton = getClientReferenceManifestSingleton()
```

Comment:
No I think it's not. That's why it's sketchy but it's not net-new broken. We should fix it for encryption too at the same time if we need to fix it. Unrelated to this pr because caches can be called in the same scopes as encrypted closures.